lasagne/updates.py

```diff
@@ -510,6 +512,56 @@ def adadelta(loss_or_grads, params, learning_rate=1.0, rho=0.95, epsilon=1e-6):
     return updates
 
 
+def adam(loss_or_grads, params, learning_rate=0.001, beta1=0.9,
+         beta2=0.999, epsilon=1e-8):
+    """Adam updates [1]_
+
+    Parameters
+    ----------
+    loss_or_grads : symbolic expression or list of expressions
+        A scalar loss expression, or a list of gradient expressions
+    params : list of shared variables
+        The variables to generate update expressions for
+    learning_rate : float
+        Learning rate
+    beta_1 : float
+        Exponential decay rate for the first moment estimates.
+    beta_2 : float
+        Exponential decay rate for the second moment estimates.
+    epsilon : float
+        Constant for numerical stability.
+
+    References
+    ----------
+    [1] Diederik Kingma, Jimmy Ba,
+    Adam: A Method for Stochastic Optimization*,
+    http://arxiv.org/abs/1412.6980
+
+    Notes
+    ----------
+    The implementation does not include lambda as it is only used to
+    prove convergence of the algorithm.
```

Comment:
As that's not explained in the paper, I'd add: "... of the algorithm (personal communication with the authors of [1]_)."